File: Monitor/lib/php/servicelib.php
<?php
function getValue($keyname)
{
    $jsonStr = getenv('JSONINPUT'); 
    $decodeJson = json_decode($jsonStr, true);
    if (isset($decodeJson[$keyname])){
        return $decodeJson[$keyname];
    }
    return "";
}

function getJsonInput()
{
    return getenv('JSONINPUT');
}

function getJsonDict()
{
    $jsonStr = getenv('JSONINPUT'); 
    return json_decode($jsonStr, true);
}

function callService($hosts, $plugin, $params)
{
    $httpCmd = "http://127.0.0.1:9090/".$plugin."?".$params."&hosts=".$hosts;
    $serviceRes = file_get_contents($httpCmd);
    return $serviceRes;
}

function getServiceValue($host, $key, $serviceRes)
{
    $decodeJson = json_decode($serviceRes, true);
    if (isset($decodeJson[$host][$key])){
        return $decodeJson[$host][$key];
    }
    return "";
}

function checkAuthority($users, $groups)
{
    $user = getenv('USERINPUT');
    $group = getenv('USERGROUP');
    $groupArr = explode(",",$group);


    $usersArr = explode(",",$users);
    if ($users != ""){
        
        if ($user!="" && in_array($user,$usersArr)){
            return true;
        }
        echo '{"status":"failed","msg":"user auth failed"}';
        exit(10);
    }

    if ($groups==""){
        return true;
    }
    $groupsArr = explode(",",$groups);
    foreach ($groupArr as $agroup){
        if (in_array($agroup,$groupsArr)){
            return true;
        }
    }
    echo '{"status":"failed","msg":"user auth failed"}';
    exit(10);
}

File: Monitor/lib/python/servicelib.py
import os
import sys
import json
import urllib2

def getValue(keyname):
    jsonStr = os.environ['JSONINPUT']
    decodeJson = json.loads(jsonStr)  
    if decodeJson.has_key(keyname):
        return decodeJson[keyname]
    return ""

def getJsonInput():
    return os.environ['JSONINPUT']

def getDictInput():
    jsonStr = os.environ['JSONINPUT']
    return json.loads(jsonStr)

def callService(hosts, plugin, params):
    httpCmd = "http://127.0.0.1:9090/"+plugin+"?"+params+"&hosts="+hosts
    try:
        page=urllib2.urlopen(httpCmd)
        line = page.read()
        page.close()
        return line
    except Exception,ex:
        return str(ex)

def getServiceValue(host, key, serviceRes):
    decodeJson = json.loads(serviceRes)
    try:
        return decodeJson[host][key]
    except Exception,ex:
        return ""
    

def checkAuthority(users, groups):
    user = os.environ['USERINPUT']
    group = os.environ['USERGROUP']
    if users != "":
        usersArr = users.split(",")
        if user!="" and user in usersArr:
            return True
        print '{"status":"failed", "msg":"user auth failed"}'
        sys.exit(10)

    if groups=="":
       return True 
    
    groupArr = group.split(",")
    groupsArr = groups.split(",")
    for agroup in groupArr:
        if agroup in groupsArr:
            return True
    print '{"status":"failed", "msg":"user auth failed"}'
    sys.exit(10)
